src/inference/elbo.ad.js

```diff
@@ -303,7 +303,12 @@ module.exports = function(env) {
       return k(s);
     },
 
-    mapDataFetch: function(data, batchSize, address) {
+    mapDataFetch: function(data, opts, address) {
+
+      var batchSize = opts.batchSize !== undefined ? opts.batchSize : data.length;
+      if (batchSize < 0 || batchSize > data.length) {
```

Comment:
Oops, sorry, I should have read the code and not just the question!

It looks like we didn't support this when it was first written, but then I [changed](https://github.com/probmods/webppl/pull/558/commits/0756b3d14f31941bbaf0c94926b7ab015644b513) it for some reason. Unfortunately, the commmit doesn't say why. I'm with you though, that this is most likely to be a mistake and we're likely better off without it. (Unless there's something we're missing.) I'll open an issue about changing this.